protocols/v1/src/methods/client_to_server.rs

```diff
@@ -18,6 +18,53 @@ use quickcheck::{Arbitrary, Gen};
 #[cfg(test)]
 use quickcheck_macros;
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct SuggestDifficulty {
+    pub id: u64,
+    pub difficulty: f64,
+}
+
+impl TryFrom<StandardRequest> for SuggestDifficulty {
+    type Error = ParsingMethodError;
+    fn try_from(msg: StandardRequest) -> Result<Self, Self::Error> {
+        match msg.params.as_array() {
+            Some(param) if param.len() == 1 => {
+                let difficulty = match &param[0] {
```

Comment:
`match` nested in another `match` makes it harder to read the code, if can think about a cleaner way to implement this, it would be great